redwood-composeui/src/commonMain/kotlin/app/cash/redwood/composeui/RedwoodContent.kt

```diff
@@ -31,13 +46,36 @@ public fun RedwoodContent(
 ) {
   val scope = rememberCoroutineScope()
   val children = remember(provider, content) { ComposeWidgetChildren() }
+
+  var viewportSize by remember { mutableStateOf(Size.Zero) }
+  val density = LocalDensity.current
+  val uiConfiguration = UiConfiguration(
+    darkMode = isSystemInDarkTheme(),
+    safeAreaInsets = safeAreaInsets(),
+    viewportSize = viewportSize,
+    density = density.density.toDouble(),
+  )
+
   LaunchedEffect(provider, content) {
     val composition = RedwoodComposition(
       scope = scope,
       container = children,
       provider = provider,
     )
-    composition.setContent(content)
+    composition.setContent {
+      CompositionLocalProvider(LocalUiConfiguration provides uiConfiguration) {
+        content()
+      }
+    }
```

Comment:
This'll be reverted as part of https://github.com/cashapp/redwood/pull/1454, where the `UiConfiguration` is provided as part of `RedwoodComposition#setContent`.